src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3850,25 +3849,10 @@ protected static boolean indexOutOfBounds(Rope rope, int byteIndex) {
 
     }
 
-    @ImportStatic(StringGuards.class)
-    public abstract static class StringAreComparableNode extends RubyBaseNode {
-
-        @Child AreComparableRopesNode areComparableRopesNode = AreComparableRopesNode.create();
-
-        public abstract boolean executeAreComparable(Object first, Object second);
-
-        @Specialization
-        protected boolean areComparable(Object a, Object b,
-                @CachedLibrary(limit = "LIBSTRING_CACHE") RubyStringLibrary libA,
-                @CachedLibrary(limit = "LIBSTRING_CACHE") RubyStringLibrary libB) {
-            return areComparableRopesNode.execute(libA.getRope(a), libB.getRope(b));
-        }
-    }
-
     @ImportStatic({ StringGuards.class, StringOperations.class })
     public abstract static class StringEqualNode extends RubyBaseNode {
 
-        @Child private StringAreComparableNode areComparableNode;
+        @Child private AreComparableRopesNode areComparableNode;
 
         public abstract boolean executeStringEqual(Object string, Object other);
```

Comment:
Should we change this so it takes 2 Ropes? and move the extraction in both callers?
Then we'd avoid RubyStringLibrary entirely in this node, and only have them in EqualNode & InlinedEqualNode (which already have them for is-it-a-string checks)

Reply:
Ok, will give that a try.